### PR TITLE
Update `Tooltip` tests for Material 3

### PR DESCRIPTION
**Pull request by TahaTesser**
Updated unit tests for `Tooltip` to have M2 and M3 versions.

More info in #139076

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/material/tooltip_test.dart [hunk 1]
```diff
@@ -198,7 +198,61 @@ void main() {
...
-  });
+  }, skip: kIsWeb && !isCanvasKit); // https://github.com/flutter/flutter/issues/99933
+
+  testWidgetsWithLeakTracking('Material3 - Does tooltip end up in the right place - top left', (WidgetTester tester) async {
+    final GlobalKey<TooltipState> tooltipKey = GlobalKey<TooltipState>();
+    late final OverlayEntry entry;
+    addTearDown(() => entry..remove()..dispose());
+
+    await tester.pumpWidget(
+      Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_test.dart:209]: Can we here replace this with MaterialApp? I know the theme data will fallback to M3 but I see another PR #139253 is trying to update the fallback value. I think using MaterialApp helps make sure this unit test is really for M3 no matter whether that PR is merged or not:)

File: packages/flutter/test/material/tooltip_test.dart [hunk 2]
```diff
@@ -438,7 +492,64 @@ void main() {
...
   });
 
-  testWidgetsWithLeakTracking('Does tooltip end up in the right place - near the edge', (WidgetTester tester) async {
+  testWidgetsWithLeakTracking('Material3 - Does tooltip end up in the right place - way off to the right', (WidgetTester tester) async {
+    final GlobalKey<TooltipState> tooltipKey = GlobalKey<TooltipState>();
+    late final OverlayEntry entry;
+    addTearDown(() => entry..remove()..dispose());
+
+    await tester.pumpWidget(
+      Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_test.dart:501]: Also here and the unit tests below. But please let me know if the comment above doesn't make sense!

File: packages/flutter/test/material/tooltip_test.dart [hunk 3]
```diff
@@ -498,6 +609,63 @@ void main() {
     expect(tip.localToGlobal(tip.size.bottomRight(Offset.zero)).dy, equals(324.0));
   });
 
+  testWidgetsWithLeakTracking('Material3 - Does tooltip end up in the right place - near the edge', (WidgetTester tester) async {
+    final GlobalKey<TooltipState> tooltipKey = GlobalKey<TooltipState>();
+    late final OverlayEntry entry;
+    addTearDown(() => entry..remove()..dispose());
+
+    await tester.pumpWidget(
+      Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_test.dart:618]: Also here.

File: packages/flutter/test/material/tooltip_test.dart (hunk 1 above)
Comment (TahaTesser) [reply packages/flutter/test/material/tooltip_test.dart]: This makes sense. Thanks for suggesting this.

File: packages/flutter/test/material/tooltip_test.dart [hunk 4]
```diff
@@ -1627,6 +1930,55 @@ void main() {
     expect(tip.size.height, equals(64.0));
   });
 
+  testWidgetsWithLeakTracking('Material3 - Tooltip text scales with textScaleFactor', (WidgetTester tester) async {
+    Widget buildApp(String text, { required double textScaleFactor }) {
+      return MediaQuery(
+        data: MediaQueryData(textScaleFactor: textScaleFactor),
+        child: Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_test.dart:1937]: nit: just a few more😄! Can we also do same changes as above:) ?

File: packages/flutter/test/material/tooltip_theme_test.dart [hunk 5]
```diff
@@ -727,7 +727,50 @@ void main() {
...
+  testWidgetsWithLeakTracking('Material3 - Tooltip decoration - ThemeData.tooltipTheme', (WidgetTester tester) async {
+    final GlobalKey<TooltipState> key = GlobalKey<TooltipState>();
+    const Decoration customDecoration = ShapeDecoration(
+      shape: StadiumBorder(),
+      color: Color(0x80800000),
+    );
+    late final OverlayEntry entry;
+    addTearDown(() => entry..remove()..dispose());
+    await tester.pumpWidget(
+      Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_theme_test.dart:739]: same here.

File: packages/flutter/test/material/tooltip_theme_test.dart [hunk 6]
```diff
@@ -771,6 +814,47 @@ void main() {
...
+    const Decoration customDecoration = ShapeDecoration(
+      shape: StadiumBorder(),
+      color: Color(0x80800000),
+    );
+
+    late final OverlayEntry entry;
+    addTearDown(() => entry..remove()..dispose());
+
+    await tester.pumpWidget(
+      Directionality(
```
Comment (QuncCccccc) [inline packages/flutter/test/material/tooltip_theme_test.dart:828]: same here.

**Review by QuncCccccc — APPROVED** [review]
LGTM! Thanks for keep doing this!

File: packages/flutter/test/material/tooltip_test.dart (hunk 4 above)
Comment (TahaTesser) [reply packages/flutter/test/material/tooltip_test.dart]: Good catch, I forgot the theme class too.

**Pull request merged**